sdk/src/packet.rs

```diff
@@ -52,8 +53,11 @@ impl Packet {
     /// Returns an immutable reference to the underlying buffer up to
     /// Packet.meta.size. The rest of the buffer is not valid to read from.
     #[inline]
-    pub fn data(&self) -> &[u8] {
-        &self.buffer[..self.meta.size]
+    pub fn data<I>(&self, index: I) -> Option<&<I as SliceIndex<[u8]>>::Output>
```

Comment:
How about keeping `data()` to avoid the need for `data(..).unwrap_or_default()` and then add an additional `data_slice()` method?

Reply:
That is exactly what I initially started to do; but then I notice all the risky places are already doing `packet.data()[x..y]`.
Very few places outside tests/benches do `data(..).unwrap_or_default()`.

I think the advantage of a leaner api is that it will be much less likely that someone later adds a `packet.data()[x..y]` again.

Reply:
I personally think your addition of a new method that accepts a `SliceIndex` will make it much less likely that anyone would misuse `data()`. Perhaps we can use this clippy lint to disallow slice indexing to be extra careful https://rust-lang.github.io/rust-clippy/master/#indexing_slicing